app/src/main/java/com/pajato/android/gamechat/database/AccountManager.java

```diff
@@ -265,19 +271,54 @@ public boolean hasAccount() {
     }
 
     /** Handle initialization by setting up the Firebase required list of registered classes. */
-    public void init(final List<String> classNameList) {
+    public void init(final AppCompatActivity context, final List<String> classNameList) {
         // Setup the class map that drives enabling and disabling Firebase. Each of these classes
         // must be registered with the app event manager before Firebase can be enabled.  If any are
         // deregistered, Firebase will effectively be disabled.
         for (String name : classNameList)
             mRegistrationClassNameMap.put(name, false);
+
+        // Save any messages necessary later
+        mMessageMap.clear();
+        mMessageMap.put(R.string.HasDepartedMessage, context.getString(R.string.HasDepartedMessage));
     }
 
     /** Return true iff the current user is a restricted/protected user. */
     public boolean isRestricted() {
         return hasAccount() && mCurrentAccount.chaperone != null;
     }
 
+    /** Remove the current account from the specified group */
+    public void leaveGroup(Group group) {
+        // Start by making sure the account is in the group
+        if (!mCurrentAccount.joinMap.keySet().contains(group.key))
+            return;
+
+        // Put a message in each room in the group, indicating that the current account has left.
+        // Then delete the account from the room's member list.
+        for (String roomKey : group.roomList) {
+            Room room = RoomManager.instance.getRoomProfile(roomKey);
+            List<String> roomMembers = room.getMemberIdList();
+            if (roomMembers.contains(mCurrentAccountKey)) {
+                String format = mMessageMap.get(R.string.HasDepartedMessage);
+                String text = String.format(Locale.getDefault(), format, mCurrentAccount.displayName);
+                MessageManager.instance.createMessage(text, STANDARD, mCurrentAccount, room);
+                RoomManager.instance.leaveRoom(room);
+            }
+        }
+
```

Comment:
You might want to collect the comments and the following code into a sort of code "paragraph".